quadra/tasks/classification.py

```diff
@@ -1069,17 +1080,8 @@ def generate_report(self) -> None:
         log.info("Generating report!")
         os.makedirs(self.report_path, exist_ok=True)
 
-        test_dataset = cast(ImageClassificationListDataset, self.datamodule.test_dataloader().dataset)
-        res = pd.DataFrame(
-            {
-                "sample": list(test_dataset.x),
-                "real_label": self.metadata["test_labels"],
-                "pred_label": self.metadata["test_results"],
-            }
-        )
-        os.makedirs(self.report_path, exist_ok=True)
         save_classification_result(
-            results=res,
+            results=self.res,
```

Comment:
Do we really need self.res here? 

Isn't this
self.metadata["test_results"] = self.res

Exactly the same? I would avoid self.res if it's not really needed